/**
 * Copyright (c) 2025 Huawei Technologies Co., Ltd.
 * This program is free software, you can redistribute it and/or modify it under the terms and conditions of 
 * CANN Open Software License Agreement Version 2.0 (the "License").
 * Please refer to the License for details. You may not use this file except in compliance with the License.
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, 
 * INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
 * See LICENSE in the root of the software repository for the full text of the License.
 */

syntax = "proto3";
package ff.udf;

message AttrValue {

    message ArrayValue {
        repeated bytes s = 2;                         //"array(string)"
        repeated int64 i = 3 [ packed = true ];       //"array(int)"
        repeated float f = 4 [ packed = true ];       //"array(float)"
        repeated bool b = 5 [ packed = true ];        //"array(bool)"
        repeated int32 type = 6 [ packed = true ];    //"array(type)"
    }

    message ListListInt{
        message ListInt{
            repeated int64 i             = 1; // array(int)
        }
        repeated ListInt list_i          = 1; // array(array(int))
    }

    oneof value {
        ArrayValue array = 1;
        bytes s = 2;           //"string"
        int64 i = 3;           //"int"
        float f = 4;           //"float"
        bool b = 5;            //"bool"
        int32 type = 6;        //"type"
        ListListInt list_list_i = 7; // array(array(int))
    }
}
